### ISSUE TITLE
Would the pre-trained model for text summarization be public also?

### ISSUE THREAD

**Issue by IreneZihuiLi** [microsoft/MASS#3]
And other tasks in the paper.

**Comment by StillKeepTry** [comment]
@IreneZihuiLi Our code and models for text summarization will also be released in the future. You can also email me to achieve model.

**Comment by sai-prasanna** [comment]
@StillKeepTry - Can you share the pre-processing code (sh files) for monoligual pre-training?

 (Weekend here, and GPUs are free to run some experiments 😅 )

**Comment by StillKeepTry** [comment]
@sai-prasanna , sorry for some delay with the response.Which kind of pre-processing script do you want?

**Comment by sai-prasanna** [comment]
Script for creating "./data/processed/en/ " given for monolingual pretraining.

> python train.py                                      \
> --exp_name mass_english                              \
> --data_path ./data/processed/en/                     \
> --lgs 'en'                                           \
> --mass_steps 'en'                                    \
> --encoder_only false                                 \
> --emb_dim 1024                                       \
> --n_layers 6                                         \
> --n_heads 8                                          \
> --dropout 0.1                                        \
> --attention_dropout 0.1                              \
> --gelu_activation true                               \
> --tokens_per_batch 3000                              \
> --optimizer adam_inverse_sqrt,beta1=0.9,beta2=0.98,lr=0.0001 \
> --epoch_size 200000                                  \
> --max_epoch 100                                      \
> --eval_bleu true                                     \
> --word_mass 0.5                                      \
> --min_len 5                                          \
> --english_only true
> "

And how to do inference on just this pre-trained model ?

**Comment by StillKeepTry** [comment]
@sai-prasanna , You means you want to test the translation quality in en->en direction?

**Comment by sai-prasanna** [comment]
@StillKeepTry yes, I want to evaluate what monolingual model pretraining has learnt, without tuning it for any new task.

**Comment by StillKeepTry** [comment]
We have released a pre-trained model for summarization task
 (include document-level) on fairseq now. You can use it from [here](https://github.com/microsoft/MASS/tree/master/MASS-summarization/).
